@xen-orchestra/web/src/components/DonutWithLegends.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="donut-with-legends">
+    <DonutChart :segments="donutSegments" :icon :max-value="maxValue" />
+    <div class="legends-and-title">
+      <slot />
+      <ul class="legends">
+        <UiLegend
+          v-for="(segment, index) in segments"
+          :key="index"
+          :color="segment.color"
+          :value="segment.value"
+          :unit="segment.unit"
+          :tooltip="segment.tooltip"
+        >
+          {{ segment.label }}
+        </UiLegend>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { DonutSegment } from '@core/types/donut-chart.type'
+import type { LegendColor } from '@core/types/ui-legend.type'
+import DonutChart from '@core/components/DonutChart.vue'
+import UiLegend from '@core/components/UiLegend.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  segments: {
+    label: string
+    value: number
+    unit?: string
+    color: LegendColor
```

Comment:
`DonutSegmentColor` being more restrictive than `LegendColor`, it would therefore make more sense to use it.

```suggestion
    color: DonutSegmentColor
```